File: backend/src/main/java/com/example/backend/model/dto/HandinDTO.java
package com.example.backend.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.time.LocalDateTime;

public class HandinDTO {
    private Long id;
    private String path;
    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
    private LocalDateTime timestamp;
    private Long commitmentId;
    private Long uploaderId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public Long getCommitmentId() {
        return commitmentId;
    }

    public void setCommitmentId(Long commitmentId) {
        this.commitmentId = commitmentId;
    }

    public Long getUploaderId() {
        return uploaderId;
    }

    public void setUploaderId(Long uploaderId) {
        this.uploaderId = uploaderId;
    }
}

File: frontend/src/app/shared/dto/handin.dto.ts
export class HandinDto {
  public readonly id: number | null;
  public readonly path: string;
  public readonly timestamp: Date;
  public readonly fileId: number;
  public readonly commitmentId: number;
  public readonly uploaderId: number;

  constructor(
    id: number | null,
    path: string,
    timestamp: Date,
    fileId: number,
    commitmentId: number,
    uploaderId: number
  ) {
    this.id = id;
    this.path = path;
    this.timestamp = timestamp;
    this.fileId = fileId;
    this.commitmentId = commitmentId;
    this.uploaderId = uploaderId;
  }
}
